### PR TITLE
Rename components

### LINKED ISSUE #21: Rename components of Safe protocol
Related to: https://github.com/5afe/safe-protocol-specs/issues/9
- [x] Component -> Integrations
- [x] Modules -> Plugins
- [x] Guards -> Hooks
- [x] Mediator -> Manager
- [x] Fallback Handler -> Function Handlers

### PR DESCRIPTION
**Pull request by akshay-ap**
PR open for comments

Relates to https://github.com/5afe/safe-protocol-specs/issues/9
Fixes #21
Changes in PR:

Bulk rename protocol components by find and replace:

 Component -> Integrations
 Modules -> Plugins
 Guards -> Hooks
 Mediator -> Manager
Fallback Handler -> Function Handlers

### CODE REVIEW

File: contracts/SafeProtocolRegistry.sol [hunk 1]
```diff
@@ -5,66 +5,66 @@ import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
 import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
 contract SafeProtocolRegistry is ISafeProtocolRegistry, Ownable2Step {
-    mapping(address => ComponentInfo) public listedComponents;
+    mapping(address => IntegrationInfo) public listedIntegrations;
 
-    struct ComponentInfo {
+    struct IntegrationInfo {
```
Comment (rmeissner) [inline contracts/SafeProtocolRegistry.sol:10]: Different topic, but should an integration have a type? Or are this only PlugIns in the first place? Then we should rename this mapping.

File: contracts/interfaces/Integrations.sol [hunk 2]
```diff
@@ -39,12 +39,12 @@ interface ISafeProtocolStaticFallbackMethod {
 }
 
 /**
- * @title ISafeProtocolGuard - An interface that a Safe guard should implement.
- * @notice In Safe protocol, a guard can approve or deny transactions based on the logic it implements.
+ * @title ISafeProtocolHook - An interface that a Safe hook should implement.
+ * @notice In Safe protocol, a hook can approve or deny transactions based on the logic it implements.
  */
-interface ISafeProtocolGuard is IERC165 {
+interface ISafeProtocolHook is IERC165 {
```
Comment (rmeissner) [inline contracts/interfaces/Integrations.sol:45]: I used the plural in the specs, as this interface specified 3 hooks.

**Review by rmeissner — DISMISSED** [review]

File: contracts/SafeProtocolRegistry.sol (hunk 1 above)
Comment (akshay-ap) [reply contracts/SafeProtocolRegistry.sol:10]: This is only for plugin.

Comment (akshay-ap) [reply contracts/SafeProtocolRegistry.sol:11]: done in #29

**Review by mmv08 — APPROVED** [review]

File: contracts/interfaces/Integrations.sol (hunk 2 above)
Comment (akshay-ap) [reply contracts/interfaces/Integrations.sol]: For now only updating to `ISafeProtocolHooks` in this PR.

**Comment by akshay-ap** [comment]
to do:
- [ ] Use `hooks` instead of `hook`

**Pull request merged**